code/data_structures/src/stack/balanced_expression/balanced_paranthesis.c

```diff
@@ -0,0 +1,60 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+typedef struct stack {
+	char data;
+	struct stack* next;
+}stack;
+
+void push(stack** top, char data) {
+    stack* new = (stack*)malloc(sizeof(stack));
+    new->data = data;
+    new->next = *top;
+    *top = new;
+}
+
+char peek(stack* top) {
+    return top->data;
+}
+
+int empty(stack* top) {
+    if(top == NULL) return 1;
+    return 0;
+}
+
+void pop(stack** top) {
+    stack* temp = *top;
+    *top = (*top)->next;
+    free(temp);
+    temp = NULL;    
+}
+
+int checkBalanced(char *s) {
+	if (strlen(s) % 2 != 0) return 0;
+	stack *head = NULL;
+	for (int i=0;i < strlen(s);i++) {
```

Comment:
Add a space around `=` and after `;`

Also, you are calling the `strlen` function at each iteration, therefore recalculating the size of the string many times, which makes the algorithm much slower. It is better to call `strlen(s)` once and store its value in a variable to compare with it.